src/main/java/com/google/sps/servlets/EventsServlet.java

```diff
@@ -0,0 +1,199 @@
+import com.google.appengine.api.datastore.Key;
+
+import com.google.appengine.api.users.UserServiceFactory;
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.User;
+
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+
+import com.google.api.client.auth.oauth2.Credential;
+import com.google.api.client.extensions.java6.auth.oauth2.AuthorizationCodeInstalledApp;
+import com.google.api.client.extensions.jetty.auth.oauth2.LocalServerReceiver;
+import com.google.api.client.googleapis.auth.oauth2.GoogleAuthorizationCodeFlow;
+import com.google.api.client.googleapis.auth.oauth2.GoogleClientSecrets;
+import com.google.api.client.googleapis.javanet.GoogleNetHttpTransport;
+import com.google.api.client.http.javanet.NetHttpTransport;
+import com.google.api.client.json.JsonFactory;
+import com.google.api.client.json.jackson2.JacksonFactory;
+import com.google.api.client.util.DateTime;
+import com.google.api.client.util.store.FileDataStoreFactory;
+import com.google.api.services.calendar.Calendar;
+import com.google.api.services.calendar.CalendarScopes;
+import com.google.api.services.calendar.model.Event;
+import com.google.api.services.calendar.model.Events;
+import com.google.api.services.calendar.model.EventDateTime;
+import com.google.api.services.calendar.model.Event.ExtendedProperties;
+
+import java.io.FileNotFoundException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.security.GeneralSecurityException;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
+import java.util.Date;
+import java.text.SimpleDateFormat;
+import java.time.*; 
+import java.util.TimeZone;
+
+/** Servlet that returns events sorted by most recent timestamp */
+@WebServlet("/events")
+public class EventsServlet extends HttpServlet {
+
+    static final String EVENT = "Event";
+    static final String TIMESTAMP = "timestamp";
+    static final String SUMMARY = "summary";
+    static final String DESCRIPTION = "description";
+    static final String LOCATION = "location";
+    static final String DATETIME = "date_time";
+    static final String CATEGORY = "category";
+    static final String UTC_TIMEZONE = "UTC";
+    static final String DATE = "date";
+    static final String START_TIME = "start";
+    static final String END_TIME = "end";
+    static final String USER_TIMEZONE = "timezone";
+
+    static final List<String> CATEGORIES = new ArrayList<String>(
+        Arrays.asList("food_beverage", "nature", "water", "waste_cleanup", "other")
+    );
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    FetchOptions fetchOptions = FetchOptions.Builder.withLimit(10);
```

Comment:
[No action needed, just a note]: You may want to handle pagination in the future by passing in a request parameter. You've already done so in your portfolio project's comments section, and the same ideas can be applied here.